app/services/date_slope.rb

```diff
@@ -0,0 +1,73 @@
+class DateSlope
+  def self.call(dates)
+    new(dates).slope
+  end
+
+  attr_reader :dates
+
+  def initialize(dates)
+    @dates = dates
+  end
+
+  def slope # rubocop:disable Metrics/AbcSize
+    1.0 * ((size * sum_xy) - (sum_x * sum_y)) / ((size * sum_xx) - (sum_x * sum_x))
+  end
+
+  def days
+    @days ||= DescriptiveStatistics::Stats.new(dates.map(&:day))
+  end
+
+  def normalized_days
+    @normalized_days ||= DescriptiveStatistics::Stats.new(days_at_beginning_bump_to_previous_month)
+  end
+
+  def best_match_days
+    return days if days.range < 10
```

Comment:
what's the meaning of `10` ?

Reply:
Good point - I've moved this (and the other 10) out to constants with comments